intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/problems/LevelRemappingPluginCreationResultResolver.kt

```diff
@@ -0,0 +1,69 @@
+package com.jetbrains.plugin.structure.intellij.problems
+
+import com.jetbrains.plugin.structure.base.plugin.PluginCreationFail
+import com.jetbrains.plugin.structure.base.plugin.PluginCreationResult
+import com.jetbrains.plugin.structure.base.plugin.PluginCreationSuccess
+import com.jetbrains.plugin.structure.base.plugin.PluginProblem
+import com.jetbrains.plugin.structure.base.problems.ReclassifiedPluginProblem
+import com.jetbrains.plugin.structure.intellij.plugin.IdePlugin
+import kotlin.reflect.KClass
+
+class LevelRemappingPluginCreationResultResolver(private val delegatedResolver: PluginCreationResultResolver,
+                                                 additionalLevelRemapping: Map<KClass<*>, PluginProblem.Level> = emptyMap()
+  ) : PluginCreationResultResolver {
+
+  private val remappedLevel: Map<KClass<*>, PluginProblem.Level> = additionalLevelRemapping +
+    mapOf(IllegalPluginIdPrefix::class to PluginProblem.Level.WARNING)
+
+  override fun resolve(plugin: IdePlugin, problems: List<PluginProblem>): PluginCreationResult<IdePlugin> {
+    return when (val pluginCreationResult = delegatedResolver.resolve(plugin, problems)) {
+      is PluginCreationSuccess -> remapSuccess(pluginCreationResult)
+      is PluginCreationFail -> remapFailure(plugin, pluginCreationResult)
+    }
+  }
+
+  private fun remapSuccess(pluginCreationResult: PluginCreationSuccess<IdePlugin>): PluginCreationResult<IdePlugin> {
+    return with(pluginCreationResult) {
+      copy(warnings = remapWarnings(warnings), unacceptableWarnings = remapUnacceptableWarnings(unacceptableWarnings))
+    }
+  }
+
+  private fun remapFailure(plugin: IdePlugin, pluginCreationResult: PluginCreationFail<IdePlugin>): PluginCreationResult<IdePlugin> {
+    return with(pluginCreationResult) {
+      val remappedErrorsAndWarnings = remapErrorsAndWarnings(errorsAndWarnings)
+      if (remappedErrorsAndWarnings.hasNoErrors()) {
+        return PluginCreationSuccess(plugin, remappedErrorsAndWarnings)
```

Comment:
Don't we need to pass warnings and unacceptable warnings separately? If I understand correctly now we pass `remappedErrorsAndWarnings` to warnings

Reply:
Constructor in `PluginCreationSuccess` will automatically categorize problems to _warnings_ and _unacceptable warnings_.

I have added an additional [unit test](https://github.com/JetBrains/intellij-plugin-verifier/pull/1008/commits/b971fc2fc8832b1afb6c088585c86b3e367f9579) to cover this.